engine/engine.go

```diff
@@ -367,8 +368,8 @@ func (e *Engine) NotifyPut(ctx context.Context, contextID []byte, md metadata.Me
 // registered.
 //
 // See: Engine.RegisterMultihashLister, Engine.Publish.
-func (e *Engine) NotifyRemove(ctx context.Context, contextID []byte) (cid.Cid, error) {
-	return e.publishAdvForIndex(ctx, contextID, metadata.Metadata{}, true)
+func (e *Engine) NotifyRemove(ctx context.Context, provider string, contextID []byte) (cid.Cid, error) {
```

Comment:
Take `Peer.ID` instead of `string` as `provider`?

Reply:
Fixed